webbpsf/detectors.py

```diff
@@ -542,3 +547,129 @@ def _show_miri_cruciform_kernel(filt, npix=101, oversample=4, detector_position=
     ax.plot(0, 0, marker='+', color='yellow')
 
     matplotlib.pyplot.colorbar(mappable=ax.images[0])
+
+# Functions for applying IFU optics systematics models
+#
+# Note, thes are is not actually a "Detector" effect, but this file is a
+# convenient place to locate that code, because similar to the detector effects
+# it's implemented as a post-processing modification on the output PSF array.
+
+
+def apply_miri_ifu_broadening(hdulist, options, slice_width=0.196):
+    """ Apply a simple empirical model of MIRI IFU broadening to better match observed PSFs
+
+    Parameters
+    -----------
+    hdulist : astropy.io.fits.HDUList
+		PSF calculation output data structure. Will be modified.
+	options : dict
+		Options dict for setting optional behaviors
+    slice_width : float
+		MIRI MRS IFU slice width (across the slice). See MIRI._IFU_pixelscale in webbpsf_core.py
+
+    """
+    # First, check an optional flag to see whether or not to include this effect.
+    # User can set the option to None to disable this step.
+    model_type = options.get('ifu_broadening', 'empirical')
+
+    if model_type is None or model_type.lower() == 'none':
+        return hdulist
+
+    ext = 1  # Apply this effect to the OVERDIST extension, which at this point in the code will be ext 1
+
+    webbpsf.webbpsf_core._log.info(f'Applying MIRI IFU broadening model: {model_type}')
+    hdulist[ext].header.add_history(f"Added broadening model for IFU PSFs: {model_type}")
+
+    hdulist[ext].header['IFUBROAD'] = (True, "IFU PSF broadening model applied")
+    hdulist[ext].header['IFUBTYPE'] = (model_type, "IFU PSF broadening model type")
+
+    if model_type.lower() == 'gaussian':
+        # Very simple model just as a Gaussian convolution kernel
+        sigma = constants.INSTRUMENT_IFU_BROADENING_PARAMETERS['MIRI']['sigma']
+        hdulist[ext].header['IFUBSIGM'] = (model_type, "[arcsec] IFU PSF broadening Gaussian sigma")
```

Comment:
For this header keyword, IFUBSIGM, it is sigma not model_type -> `(sigma, "[arcsec] IFU PSF broadening Gaussian sigma")`